File: VotingModelLibrary/Models/Race.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace VotingModelLibrary.Models
{
    public class Race
    {
        [Key]
        [Display(Name = "RaceId")]
        public int RaceId { get; set; }
        [Display(Name = "ElectionId")]
        public int ElectionId { get; set; }
        [Display(Name = "Election")]
        public Election Election { get; set; }
        [Display(Name = "PositionName")]
        public string PositionName { get; set; }
        [Display(Name = "NumberNeeded")]
        public int NumberNeeded { get; set; }
        [Display(Name = "CandidateRaces")]
        public List<CandidateRace> CandidateRaces { get; set; }
    }
}

File: Spa/ClientApp/src/app/models/Race.ts
import { CandidateRace } from './CandidateRace';
import { Candidate } from './candidate';

export class Race {
  raceId : number;
  electionId: number;
  election: string;
  positionName : string;
  numberNeeded: number;
  candidateRaces: CandidateRace[];
  show: string;
  selected: Candidate[];

  constructor(
    raceId: number,
    electionId: number,
    election: string,
    positionName: string,
    numberNeeded: number,
    candidateRaces: CandidateRace[],
    show: string
  ) {
    this.raceId = raceId;
    this.electionId = electionId;
    this.election = election;
    this.positionName = positionName;
    this.numberNeeded = numberNeeded;
    this.candidateRaces = candidateRaces;
    this.show = "true";
    this.selected = [];

  }
}
